modules/reindex/src/main/java/org/elasticsearch/index/reindex/RestReindexAction.java

```diff
@@ -150,11 +150,13 @@ static RemoteInfo buildRemoteInfo(Map<String, Object> source) throws IOException
         String host = hostMatcher.group("host");
         int port = Integer.parseInt(hostMatcher.group("port"));
         Map<String, String> headers = extractStringStringMap(remote, "headers");
+        String timeoutString = extractString(remote, "timeout");
+        TimeValue timeout = timeoutString == null ? RemoteInfo.DEAULT_SOCKET_TIMEOUT : parseTimeValue(timeoutString, "timeout");
```

Comment:
should we call the field `socket_timeout` ? Will we want to add `connect_timeout` too in the future? I think timeout is very generic so we may want to be more specific.